File: csharp/resistor-color/ResistorColor.cs

public static class ResistorColor
{
    private static readonly string[] colors = {
        "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
    };

    public static int ColorCode(string color)
    {
        for(int index = 0; index < colors.Length; index++)
        {
            if (color == colors[index])
            {
                return index;
            }
        }

        return -1;
    }

    public static string[] Colors() => colors;
}

File: typescript/resistor-color/resistor-color.ts
export enum Colors {
  black,
  brown,
  red,
  orange,
  yellow,
  green,
  blue,
  violet,
  grey,
  white,
}

type Color = keyof typeof Colors

export const colorCode = (color: Color): number => {
  return (<any>Colors)[color];
}

export const COLORS = Object.keys(Colors)
                        .filter(key => typeof Colors[key as any] === 'number');
